core/rawdb/freezer_table.go

```
@@ -349,12 +349,15 @@ func (t *freezerTable) releaseFile(num uint32) {
 	}
 }
 
-// releaseFilesAfter closes all open files with a higher number
-func (t *freezerTable) releaseFilesAfter(num uint32) {
+// releaseFilesAfter closes all open files with a higher number, and optionally also deletes the files
+func (t *freezerTable) releaseFilesAfter(num uint32, remove bool) {
 	for fnum, f := range t.files {
 		if fnum > num {
 			delete(t.files, fnum)
 			f.Close()
+			if remove {
+				os.Remove(f.Name())
```

We probably want to handle any failure here and return an error from the method, otherwise we might end up in a really weird state if someone does a SetHead without write permissions on the files.

---

Mmmno, actually, it doesn't matter if we succeed or not. It's just a courtesy against the user, to free up some space. Before I added this, we didn't bother, and that works too. 
We still open it in truncate-mode later, and if user can't write, then something will blow up. But this isn't necessarily the place to catch it

---

Ah, fair enough. Perhaps add a comment why we ignore the error? It's not that obvious, whereas deleting a file and ignoring the failure looks funky :)